### ISSUE TITLE
Tags

### ISSUE THREAD

**Issue by drewroberts** [drewroberts/media#5]
Tags are used for YouTube (#4) Videos (#2). They are pulled and saved. Unlike YouTube, we will not allow spaces with our tags. They will also be forced to have a # sign in front of them.

**Comment by drewroberts** [comment]
Tags will have a Many to Many relationship with Videos.

**Comment by drewroberts** [comment]
Would also like to use this style for Nova:

https://github.com/spatie/nova-tags-field

**Comment by drewroberts** [comment]
Here's an example of a good Tags implementation:

- https://github.com/spatie/laravel-tags

The difference that I want is that we should strip the Name of all spaces, dashes, underscores, and other characters and should add the hashtag to the beginning. It should be accessible with or without the hashtag and there should be a check to make sure it is unique regardless of capitalization although I want to make sure to allow caps wherever when it is saved for the first time - Ex: #DrewRoberts or #RollTide

The slug should be without the hashtag and all lowercase.

I also prefer to use a string field instead of json like that package.

**Comment by gurgentil** [comment]
@drewroberts Do you need all those methods in `spatie/laravel-tags`, some of them are just slight variations of other ones.

**Comment by gurgentil** [comment]
@drewroberts can we add the methods (and test them) as we need them?

**Comment by gurgentil** [comment]
@drewroberts about the hashtag, I think it would be better if we only used the hashtag when parsing strings. Is there an specific reason why you want to persist them? I feel like, if all tags have hashtags, there is no point in storing them. Add them to the API transformer or simply prepend them when they need to be rendered.

**Comment by drewroberts** [comment]
No, we don't need all the methods. I am fine with adding and testing them as we need them.

I like using the hashtag at the beginning and agree that it probably doesn't need to be saved. It might be better though for the way people are searching and accessing hashtags to not always try to add them on.

**Comment by gurgentil** [comment]
@drewroberts When you say "It should be accessible with or without the hashtag". What do you mean by accessible?
